Specs/Renderer/ShaderProgramSpec.js

```diff
@@ -355,6 +358,27 @@ defineSuite([
         expect(renderFragment(context, sp)).toEqual([255, 255, 255, 255]);
     });
 
+    it('creates duplicate uniforms if precision of uniforms in vertex and fragment shader do not match', function() {
```

Comment:
Ah, I see now.  No, one test that covers both manual and automatic uniforms is fine (especially now that I looked at your implementation).  However, my comment about the fragile test still stands.